aten/src/ATen/cuda/CublasHandlePool.cpp

```diff
@@ -41,31 +41,38 @@ void clearCublasWorkspaces() {
 
 size_t parseChosenWorkspaceSize() {
   const char * val = getenv("CUBLAS_WORKSPACE_CONFIG");
+  /* :4096:2:16:8 default, 32MiB for Hopper */
+  int device;
+  cudaDeviceProp* properties = at::cuda::getDeviceProperties(cudaGetDevice(&device));
+  const bool sm90 = properties->major == 9 && properties->minor == 0;
```

Comment:
Defensive programming
```suggestion
  const bool sm90 = properties != nullptr && properties->major == 9 && properties->minor == 0;
```